cedar-policy-core/src/parser.rs

```diff
@@ -1146,4 +1141,41 @@ mod parse_tests {
             "expected `!=`, `&&`, `(`, `*`, `+`, `-`, `.`, `::`, `<`, `<=`, `==`, `>`, `>=`, `[`, `||`, `has`, `in`, `is`, `like`, or `then`",
         )
     }
+
+    #[test]
+    fn string_escapes() {
+        // test strings with valid escapes
+        // convert a string `s` to `<double-quote> <escaped-form-of-s> <double-quote>`
+        // and test if the resulting string literal AST contains exactly `s`
+        // for instance, "\u{1F408}"" is converted into r#""\u{1F408}""#,
+        // the latter should be parsed into `Literal(String("🐈"))` and
+        // `🐈` is represented by '\u{1F408}'
+        let test_valid = |s: &str| {
+            let r = parse_literal(&format!("\"{}\"", s.escape_default()));
+            assert!(r.is_ok());
+            assert_eq!(r.unwrap(), ast::Literal::String(s.into()));
```

Comment:
While we're in here, this should give a nicer error. 

```suggestion
            assert_eq!(r.unwrap(), Ok(ast::Literal::String(s.into())));
```

Reply:
I think you need to remove the `.unwrap()` as well